src/post/api/getPosts.ts

```diff
@@ -1,9 +1,23 @@
 import { Post } from "../types";
 
-const getPosts = async (): Promise<Post> => {
+const getPosts = async (): Promise<Post[]> => {
   const apiResponse = await fetch(`${import.meta.env.VITE_API_URL}posts`);
 
-  return apiResponse.json();
+  const apiPostsDto = (await apiResponse.json()) as Post[];
```

Comment:
No se puede llamar `apiPostsDto` a un valor que habéis tipado como `Post[]`, y no estáis usando el tipado del DTO en ningún sitio.

Reply:
Está corregido